pennylane_lightning/lightning_qubit/_measurements.py

```diff
@@ -133,6 +138,62 @@ def expval(self, measurementprocess: MeasurementProcess):
             measurementprocess.obs.name, measurementprocess.obs.wires
         )
 
+    # pylint: disable=protected-access
+    def probs(self, measurementprocess: MeasurementProcess):
+        """Probabilities of the supplied observable or wires contained in the MeasurementProcess.
+
+        Args:
+            measurementprocess (StateMeasurement): measurement to apply to the state
+
+        Returns:
+            Probabilities of the supplied observable or wires
+        """
+        diagonalizing_gates = measurementprocess.diagonalizing_gates()
+        if diagonalizing_gates:
+            self._qubit_state.apply_operations(diagonalizing_gates)
+        results = self._measurement_lightning.probs(measurementprocess.wires.tolist())
+        if diagonalizing_gates:
+            self._qubit_state.apply_operations(
+                [qml.adjoint(g) for g in reversed(diagonalizing_gates)]
```

Comment:
```suggestion
                [qml.adjoint(g, lazy=False) for g in reversed(diagonalizing_gates)]
```
This is more likely to give another "native" operation, instead of a nested operation class. For example, `qml.adjoint(qml.Hadamard(0), lazy=False)` gives `Hadamard(0)`, not `Adjoint(Hadamard(0))`.